### PR TITLE
Bump PHC to 4.2.1

### PR DESCRIPTION
**Pull request by tonidero**
This PR bumps PHC to 4.2.1. As part of that change, we have added `TransactionIdentifier` and `ProductIdentifier` to `StoreTransaction` while deprecating the old `RevenueCatId` and `ProductId`.

This PR also contains a couple of small fixes for the release automation process.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -80,7 +80,7 @@ lane :update_hybrid_common_versions do |options|
   UI.message("ℹ️  Setting Purchases Hybrid Common version: #{new_version_number}")
   files_to_update = [
     'RevenueCat/Plugins/Editor/RevenueCatDependencies.xml',
-    'Subtester/Assets/RevenueCat/Plugins/Editor/RevenueCatDependencies.xml',
```
Comment (aboedo) [inline fastlane/Fastfile:83]: was this one duplicated?

File: RevenueCat/Scripts/StoreTransaction.cs [hunk 2]
```diff
@@ -17,13 +17,31 @@ public class StoreTransaction
...
+        [Obsolete("Deprecated, use TransactionIdentifier instead.", false)]
         public readonly string RevenueCatId;
+
+        /**
+         * <summary>
+         * Id associated with the transaction in RevenueCat.
+         * </summary>
+         */
+        public readonly string TransactionIdentifier;
+
```
Comment (aboedo) [inline RevenueCat/Scripts/StoreTransaction.cs:29]: Ohh I see here's where the deprecation happened

Comment (aboedo) [reply RevenueCat/Scripts/StoreTransaction.cs:29]: @vegaro had looked at these fields in the past, just tagging him to ensure this is what he had in mind for unifying

File: fastlane/Fastfile (hunk 1 above)
Comment (tonidero) [reply fastlane/Fastfile:83]: Actually that was a symlink to the one above, so yeah kinda duplicated. The new one is one I missed on the first PR.

**Review by vegaro — APPROVED** [review]

**Pull request merged**